### PR TITLE
Implemented new changes

### PR DESCRIPTION
**Pull request by Shambhavikumari**
oAuth-token-based-Authentication before subscribing to a message already available in the broker.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@Shambhavikumari Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-web-mqtt&pullRequestId=63)!

[Click here](https://cla.pivotal.io/sync/rabbitmq?repositoryId=rabbitmq/rabbitmq-web-mqtt&pullRequestId=63) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: src/rabbit_web_mqtt_connection_sup.erl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 %% The contents of this file are subject to the Mozilla Public License
 %% Version 1.1 (the "License"); you may not use this file except in
 %% compliance with the License. You may obtain a copy of the License
-%% at https://www.mozilla.org/MPL/
+%% at http://www.mozilla.org/MPL/
```
Comment (michaelklishin) [inline src/rabbit_web_mqtt_connection_sup.erl]: Please undo all changes not related to the problem at hand.

File: src/rabbit_web_mqtt_handler.erl [hunk 2]
```diff
@@ -74,10 +75,11 @@ init(Req, Opts) ->
 
 websocket_init(State = #state{conn_name = ConnStr, socket = Sock, peername = PeerAddr}) ->
     rabbit_log_connection:info("accepting Web MQTT connection ~p (~s)~n", [self(), ConnStr]),
+	rabbit_log_connection:info("rabbit_web_mqtt_connection_handler websocket_init"),
     AdapterInfo = amqp_connection:socket_adapter_info(Sock, {'Web MQTT', "N/A"}),
-    RealSocket = rabbit_net:unwrap_socket(Sock),
-    ProcessorState = rabbit_mqtt_processor:initial_state(Sock,
-        rabbit_mqtt_reader:ssl_login_name(RealSocket),
+  	%  RealSocket = rabbit_net:unwrap_socket(Sock),
+    ProcessorState = rabbit_mqtt_processor_custom:initial_state(Sock,
```
Comment (michaelklishin) [inline src/rabbit_web_mqtt_handler.erl:81]: I don't think this module exists in the MQTT plugin.

File: src/rabbit_web_mqtt_handler.erl [hunk 3]
```diff
@@ -74,10 +75,11 @@ init(Req, Opts) ->
...
+        rabbit_mqtt_reader: ssl_login_custom_subscriber(),
```
Comment (michaelklishin) [inline src/rabbit_web_mqtt_handler.erl:82]: I don't think this function exists in the MQTT plugin?

File: src/rabbit_web_mqtt_app.erl [hunk 4]
```diff
@@ -66,17 +66,10 @@ mqtt_init() ->
...
-    max_connections => get_max_connections(),
-    num_acceptors => get_env(num_tcp_acceptors, 10)
-  },
-  case ranch:start_listener(web_mqtt,
-                            ranch_tcp,
-                            RanchTransportOpts,
-                            rabbit_web_mqtt_connection_sup,
-                            CowboyOpts) of
+  MaxConnections = get_max_connections(),
+  case ranch:start_listener(web_mqtt, get_env(num_tcp_acceptors, 10),
```
Comment (michaelklishin) [inline src/rabbit_web_mqtt_app.erl:70]: What's the reasoning behind moving away from map-based transport options here?

**Review by michaelklishin — CHANGES_REQUESTED** [review]
Thanks.

Are you sure this PR was meant to be sent to this repository and not a custom fork of yours or something like that?

It would be nice to have some justification for the changes, including a problem statement, answers to a few questions in this review and a set of steps for us to reproduce the problem and see the effects of this change.

**Comment by michaelklishin** [comment]
[rabbitmq-users thread](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/rabbitmq-users/uF05pvy8hk8/WhMCYZHlAAAJ) with some background.

File: src/rabbit_web_mqtt_handler.erl (hunk 2 above)
Comment (michaelklishin) [reply src/rabbit_web_mqtt_handler.erl:81]: Also, the name is not very specific. How exactly is "custom" different from a regular processor module? When should it be used? Why? The name raises a bunch of questions and answers none.

**Comment by michaelklishin** [comment]
Instead of adding reverts or manual reverts to this PR create a new branch with only the relevant changes, and submit that with any dependencies necessary (this PR references modules that do not exist anywhere) some additional information requested above:

 * How do other RabbitMQ users benefit from this change?
 * How does it fit into the existing authN/authZ framework?
 * if it adds additional authorization checks to a certain operation, why can't that be achieved using `rabbitmq_auth_backend_http`, a custom HTTP[S]-based service, and `rabbitmq_auth_backend_cache` to mitigate the overhead? Do we really need to modify this plugin and the MQTT one?
 * How do we test it?

**Pull request closed without merging**